src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -860,6 +931,34 @@ protected theorem extractLsb_ofNat (x n : Nat) (hi lo : Nat) :
     getLsbD (extractLsb hi lo x) i = (i ≤ (hi-lo) && getLsbD x (lo+i)) := by
   simp [getLsbD, Nat.lt_succ]
 
+@[simp] theorem getLsbD_extractLsb {hi lo : Nat} {x : BitVec n} {i : Nat} :
+    getLsbD (extractLsb hi lo x) i = (decide (i < hi - lo + 1) && x.getLsbD (lo + i)) := by
+  rw [extractLsb, getLsbD_extractLsb']
+
+@[simp] theorem getMsbD_extractLsb {hi lo : Nat} {x : BitVec w} {i : Nat} :
+    (extractLsb hi lo x).getMsbD i =
+      (decide (i < hi - lo + 1) &&
+      (decide (max hi lo - i < w) &&
+      x.getMsbD (w - 1 - (max hi lo - i)))) := by
```

Comment:
```suggestion
@[simp] theorem getMsbD_extractLsb {hi lo : Nat} {x : BitVec w} {i : Nat} :
    (extractLsb hi lo x).getMsbD i =
      (decide (i < hi - lo + 1) &&
      (decide (max hi lo - i < w) &&
      x.getMsbD (w - 1 - (max hi lo - i)))) := by
```

I want to come back to this and thik a bit more about whether the RHS can be rephrased to avoid *two* conditions on the index. Maybe we can't, but I'd like to convince myself of the fact that we can't :)

Reply:
Right. The getMsbD also implies some bounds. E.g., `w - 1 - (max hi lo - i) < w`, but the issue is that the expression `w - 1 - (max hi lo - i)` saturates at zero. The `max hi lo - i < w` avoids this saturation. I would be curious if this is somewhere else already implied.

Reply:
my understanding is that having less than two conditions is not possible because we want to check 
1. out-of-bound condition wrt. extracted subvector, i.e., `i` must be within the bounds of the extracted sub-bitvec (whose length is `hi - lo + 1`)
2. out-of-bound condition wrt. original subvector, i.e., since `extractLsb` can extend the original vector, in case `w <= hi`, i.e., we need to know whether the `i` we're looking for belongs in these zero-extended bits or is within the original bv. An additional layer is that in this case we can have `lo >= hi`, and this yields a bitvector with length one. 
afaiu, trying to put these together always eventually yields to (at least) one extra condition.

Reply:
Right, I believe this. In some sense, `getLsb` allows us to not have to consider indexing with values like `-1`. But when we switch the direction to `getMsb`, we are implicitly adding a condition to handle this case of reading zeroes toward the *right*. We get another condition, as explained, to handle the out of bounds with respect to the parent bitvector, where we read zeroes from the *left*.